consensus/spos/bn/subroundEndRound.go

```diff
@@ -82,6 +82,12 @@ func (sr *subroundEndRound) doEndRoundJob() bool {
 	sr.Header.SetSignature(sig)
 
 	timeBefore := time.Now()
+	prevHeader := sr.Blockchain().GetCurrentBlockHeader()
```

Comment:
I think consensus should know only about a single processor, he should not care about how many there are. PeerProcessor could be under BlockProcessor.

Reply:
Done, moved to block processor